gcp/modules/gke/main.tf

```diff
@@ -0,0 +1,70 @@
+# copyright 2020 Datastax LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+resource "google_container_cluster" "container_cluster" {
+  name        = var.name
+  project     = var.project_id
+  description = "Demo GKE Cluster"
+  location    = var.location
+
+  remove_default_node_pool = true
+  initial_node_count       = var.initial_node_count
+
+  network    = var.network_link
+  subnetwork = var.subnetwork_link
+
+  master_auth {
+    # Setting an empty username and password explicitly disables basic auth
+    username = ""
+    password = ""
+
+    # Whether client certificate authorization is enabled for this cluster.
+    client_certificate_config {
+      issue_client_certificate = false
+    }
+  }
+
+  provisioner "local-exec" {
+    command = "sleep 120"
+  }
+
+  provisioner "local-exec" {
+    command = format("gcloud container clusters get-credentials %s --region %s --project %s", google_container_cluster.container_cluster.name, google_container_cluster.container_cluster.location, var.project_id)
+  }
+
+}
+
+
+resource "google_container_node_pool" "container_node_pool" {
+  name       = format("%s-node-pool", var.name)
+  project    = var.project_id
+  location   = var.location
+  cluster    = google_container_cluster.container_cluster.name
+  node_count = 1
+
+  node_config {
+    preemptible  = true
+    machine_type = var.machine_type
+
+    metadata = {
+      disable-legacy-endpoints = "true"
+    }
+
+    service_account = var.service_account
+    oauth_scopes = [
+      "https://www.googleapis.com/auth/logging.write",
```

Comment:
I am guessing that the operators will need more permissions, specifically for backups.  We probably will want to use https://cloud.google.com/kubernetes-engine/docs/how-to/hardening-your-cluster#workload_identity

Also, take a look at that document and ensure that we are following any of the steps that are easy.  The private API will not be easy, but we should look at that as an option.